artiq/applets/big_number.py

```diff
@@ -26,47 +27,78 @@ def __init__(self, args, ctl):
         QtWidgets.QStackedWidget.__init__(self)
         self.dataset_name = args.dataset
         self.ctl = ctl
+        self.value = 0
+        self.metadata = {}
 
-        self.lcd_widget = QResponsiveLCDNumber()
-        self.lcd_widget.setDigitCount(args.digit_count)
-        self.lcd_widget.doubleClicked.connect(self.start_edit)
-        self.addWidget(self.lcd_widget)
+        self.display_widget = QResponsiveLabel()
+        self.display_widget.setAlignment(
+            QtCore.Qt.AlignRight | QtCore.Qt.AlignVCenter)
+        self.display_widget.doubleClicked.connect(self.start_edit)
+        self.addWidget(self.display_widget)
 
         self.edit_widget = QCancellableLineEdit()
-        self.edit_widget.setValidator(QtGui.QDoubleValidator())
-        self.edit_widget.setAlignment(QtCore.Qt.AlignRight)
+        self.edit_widget.setAlignment(
+            QtCore.Qt.AlignRight | QtCore.Qt.AlignVCenter)
         self.edit_widget.editCancelled.connect(self.cancel_edit)
         self.edit_widget.returnPressed.connect(self.confirm_edit)
         self.addWidget(self.edit_widget)
 
         font = QtGui.QFont()
         font.setPointSize(60)
+        self.display_widget.setFont(font)
         self.edit_widget.setFont(font)
 
-        self.setCurrentWidget(self.lcd_widget)
+        self.setCurrentWidget(self.display_widget)
+
+    def get_unscaled_format(self):
+        scale = self.metadata.get("scale", None)
+        unit = self.metadata.get("unit", "")
+        unit_scale = scale_from_metadata({"unit": unit})
```

Comment:
can just pass self.metadata straight into scale_from_metadata and divide by it I believe.